src/update/UpdateManager.cpp

```diff
@@ -130,6 +130,42 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
 
                 if (code == wxID_YES)
                 {
+                    const audacity::network_manager::Request downloadRequest(mVersionPatch.download.ToStdString());
+                    auto downloadResponse = audacity::network_manager::NetworkManager::GetInstance().doGet(downloadRequest);
+
+                    downloadResponse->setRequestFinishedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
```

Comment:
- is downloadReponse really needed in the lambda capture?
- Are you sure this lambda executes in the main thread?  (try a breakpoint). If not, are you sure it is safe to destroy the dialog?  If not, then consider CallAfter.
- But then, mProgressDialog should not be a unique_ptr.  It should be a wxWindowPtr<>.  Remember that wxWindow objects have special memory management and should not be handled with the usual C++ smart pointers.
- https://docs.wxwidgets.org/3.0/classwx_window_ptr_3_01_t_01_4.html